### DISCUSSION TITLE
Is there any support for Mutation operation in dynamic schemas?

### DISCUSSION THREAD

**Discussion by gianalarcon** [async-graphql/async-graphql#1338] [Q&A] [answered]
So far I checked the code and examples about static and dynamic schema.
For **dynamics schema** there is a example(starwars) with the **poem** framework that shows how to build dynamic schema but it only uses query operation. Then In the dynamic documentation I found a simple example how to use subscription operation in dynamic schema. But I could not find anything related how to implement **mutation** operation in dynamic schema

I want to make sure if the actual version of async-graphql supports mutation operation in dynamic schema. If it supports, can you show me an example?

Thank you so much. The projects that you are building is amazing!

**Comment by sunli829** [comment, marked as answer]
```rust
     let mutation = Object::new("Mutation").field(
            Field::new("update", TypeRef::named(TypeRef::Bool), |_| {
                // do update
            })
            .argument(InputValue::new("id", TypeRef::STRING))
            .argument(InputValue::new("value", TypeRef::INT)),
        );
        let schema = Schema::build("Query", Some("Mutation"), None)
            .register(query)
            .register(mutation)
            .finish()
            .unwrap();
```

**Reply by gianalarcon** [reply]
Thanks for your kindly response. I am on my way to implement the **dynamic schema version** of your example **poem-subscription**(books). Hope you can help me in case I face some issues. Thank you
